### DISCUSSION TITLE
Best way to handle max open trades on a trading system

### DISCUSSION THREAD

**Discussion by gaugau3000** [nautechsystems/nautilus_trader#788] [Q&A] [answered]
Hi,

Given I have a strategy that use limit orders and is run on 50 markets 
and i limit the max open trades open at the same time to 3 in a config variable.
When i have 10 entry signals that come on the same candle (on_bar) and i try to filter using the cache orders_open + orders_inflight < max_open_trades
Then the cache seems to have a delay and i end with more open trades trades than i expected.

Is it the expected behavior (process are independents and so the time to write the cache for orders_inflight the info can be outdated of a few ms) ? Do you recommend another way to do such a thing ?

Many tks !!

**Comment by cjdsellers** [comment, marked as answer]
It helps to consider the event driven nature of the platform. For live trading there will be a delay between submitting orders and events based on that coming back into the system.

Even for backtesting the `SimulatedExchange` simulates the processing of trading messages.

And so I would carefully consider your strategy logic, and leverage the `on_event` method to take actions based on event triggers, with no assumptions as to the state the cache may be in after executing commands, if that makes sense?
